python/pyspark/pandas/indexes/timedelta.py

```diff
@@ -108,7 +109,7 @@ def __new__(
                 FutureWarning,
             )
         if not is_hashable(name):
-            raise TypeError("Index.name must be a hashable type")
+            raise PySparkTypeError(message="Index.name must be a hashable type")
```

Comment:
@HyukjinKwon @itholic Adding `message` only will reserve consistency with native Pandas, but might not be consistent with pyspark.sql, which has error_class and error_parameter. WDYT?